packages/toc/src/utils/markdown.ts

```diff
@@ -73,14 +74,25 @@ export function getHeadings(text: string): IMarkdownHeading[] {
       continue;
     }
 
-    // Don't check for Markdown headings if in a code block:
+    // Don't check for Markdown headings if in a code block
     if (line.startsWith('```')) {
       isCodeBlock = !isCodeBlock;
     }
     if (isCodeBlock) {
       continue;
     }
 
+    // Don't check for Markdown headings if in a YAML frontmatter block.
+    // We can only start a frontmatter block on the first line of the file.
+    // At other positions in a markdown file, '---' represents a horizontal rule.
+    if (line === '---' && (isFrontmatterBlock || lineIdx === 0)) {
```

Comment:
Could we move this logic before the for loop line 69? Then you can enter the _regular_ parsing with `lineIdx` greater than 0 (aka after the header).

So the code will look like:

```typescript
// Skip frontmatter header if any

// ...

for (let lineIdx = lineAfterHeader; lineIdx < lines.length; lineIdx++) {
  // parsing for headings
}